compair/static/modules/image-viewer/image-viewer-directive.js

```diff
@@ -0,0 +1,273 @@
+                    var displayWidth = img.width;
+                    var parent = $(canvas.parentNode);
+
+                    if (Math.abs(scope.orientation.rotate) == 90 ||
+                        Math.abs(scope.orientation.rotate) == 270) {
+                        ratio = 1 / ratio;
+                        displayWidth = img.height;
+                    }
+                    displayWidth = Math.min(displayWidth, parent.width());
+
+                    // scale down the image to fit if necessary
+                    scope.orientation.zoom = Math.min(getMaxZoom(img, displayWidth, scope.orientation.rotate), 1);
+
+                    scope.$applyAsync(function() {
+                        canvas.width  = displayWidth;
+                        canvas.height = canvas.width / ratio;
+                        centerImg();
+                        applyTransform();
+                    });
+                };
+
+                // return a data URL of the original image with viewing rotation
+                var rotateOriginalImage = function(mimeType) {
+                    // use an invisible canvas to rotate the original image
+                    var tempCanvas = document.createElement("canvas");
+                    var tempCtx = tempCanvas.getContext("2d");
+
+                    if (Math.abs(scope.orientation.rotate) == 90 ||
+                        Math.abs(scope.orientation.rotate) == 270) {
+                        tempCanvas.width = img.height;
+                        tempCanvas.height = img.width;
+                    } else {
+                        tempCanvas.width = img.width;
+                        tempCanvas.height = img.height;
+                    }
+
+                    var centerX=tempCanvas.width/2, centerY=tempCanvas.height/2;
+
+                    tempCtx.save();
+                    tempCtx.translate(centerX, centerY);
+                    tempCtx.rotate(scope.orientation.rotate * Math.PI/180);
+                    if (Math.abs(scope.orientation.rotate) == 90 ||
+                        Math.abs(scope.orientation.rotate) == 270) {
+                        tempCtx.translate(-centerY, -centerX);
+                    } else {
+                        tempCtx.translate(-centerX, -centerY);
+                    }
+                    tempCtx.drawImage(img, 0, 0, img.width, img.height)
+                    tempCtx.restore();
+
+                    return tempCanvas.toDataURL(mimeType);
+                };
+
+                angular.element($window).bind('resize', function() {
+                    scope.$apply();
+                });
+
+                var refreshUploadFile = function() {
+                    // if uploadFile is given, replace it with a new File object in current orientation
+                    if (scope.uploadFile) {
+                        return dataUrlToFile(rotateOriginalImage(uploadFileMimeType),
+                            scope.uploadFile.name? scope.uploadFile.name : 'image.png',
```

Comment:
Think it would be worth adding this to a easily access variable like mimetype (then no need to pass to functions)